x/rollup/keeper/msg_server.go

```diff
@@ -227,3 +229,24 @@ func (k *Keeper) SetL1BlockHistory(ctx context.Context, info *derive.L1BlockInfo
 func evmToCosmos(addr common.Address) sdk.AccAddress {
 	return addr.Bytes()
 }
+
+// TODO: This is a temporary change while the rollup module refactor is being done. Change when possible
+func ValidateBasic(m *rollupv1.InitiateWithdrawalRequest) error {
+	// We should ensure that the target field is a valid Ethereum address and that the gas_limit and data field are valid.
+	if common.IsHexAddress(m.Target) {
+		return fmt.Errorf("invalid Ethereum address: %s", m.Target)
+	}
+	// Ensure gas_limit is within a reasonable range
+	// https://insights.deribit.com/industry/ethereums-gas-mechanics/
+	gasLimit := binary.BigEndian.Uint64(m.GasLimit) // size=24 (0x18), offset=64 (0x40)
+	if gasLimit < 21000 || gasLimit > 15000000 {
+		return fmt.Errorf("gas limit must be between 21,000 and 15,000,000: %d", gasLimit)
+	}
+	// The data field should be in valid hex format
+	_, err := hex.Decode([]byte{}, m.Data)
```

Comment:
Were you able to verify that the `L2ToL1MessagePasser` and `OptimismPortal` contracts expect data to always be hex? Could it have a different encoding instead and still work? I'm curious if we need to validate anything here, it's probably worth checking to see how this is handled in Solidity

Reply:
As I can see, we don't need to validate date field. 
[Here](https://docs.optimism.io/stack/protocol/rollup/withdrawal-flow) is information about data field in withdrawal flows. 
[Here](https://github.com/ethereum-optimism/optimism/blob/62c7f3b05a70027b30054d4c8974f44000606fb7/packages/contracts-bedrock/contracts/L2/L2ToL1MessagePasser.sol#L98) we have initiateWithdrawal function.
There are no requirements for data field.
If you know more about data flow after this function, please share.